frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2828,9 +2828,11 @@ helpResolveFunction(ResolutionContext* rc, const TypedFnSignature* sig,
   // same function twice when working with inferred 'out' formals)
   sig = sig->inferredFrom();
 
-  if (!sig->isInitializer() && sig->needsInstantiation()) {
-    CHPL_ASSERT(false && "Should only be called on concrete or fully "
-                         "instantiated functions");
+  if (!sig->isInitializer() && !sig->untyped()->isTypeConstructor() &&
+      sig->needsInstantiation()) {
```

Comment:
This is the

> per bug fix suggestion from Ben

part of the PR, but could you explain why it's needed? For type constructors that accept generic types? A case in a comment would help.

Reply:
Will comment. It's needed for type constructors accepting fully-generic types, at least -- `var r : range(?);` hits this assert without it. (I had said in offline conversation that it didn't anymore, but I just reproduced it now, so I must have made a mistake.)

Reply:
This particular use surprises me. Is `(?)` passed as a special argument to `range`? I would not have expected that, in favor of `hasQuestionArg`. And I certainly would expect `range(?)` to simply be the fully generic type (not an instantiation).

Reply:
Nevermind, I read your comment and I'm big dumb. All clear.